pkg/cli/login/login.go

```diff
@@ -88,7 +91,12 @@ func NewCmdLogin(f kcmdutil.Factory, streams genericiooptions.IOStreams) *cobra.
 	cmds.Flags().StringVarP(&o.Password, "password", "p", o.Password, "Password for server")
 
 	cmds.Flags().BoolVarP(&o.WebLogin, "web", "w", o.WebLogin, "Login with web browser. Starts a local HTTP callback server to perform the OAuth2 Authorization Code Grant flow. Use with caution on multi-user systems, as the server's port will be open to all users.")
-	cmds.Flags().Int32VarP(&o.CallbackPort, "callback-port", "c", o.CallbackPort, "Port for the callback server when using --web. Defaults to a random open port")
+	cmds.Flags().Int32VarP(&o.CallbackPort, "callback-port", "c", o.CallbackPort, "Port for the callback server when using --web or --exec-plugin. Defaults to a random open port if used with --web. Defaults to 8000 if used with --exec-plugin")
```

Comment:
Wrong description (read it all).
Default to a random unused port unless there is a good reason not to do so.

Reply:
Keycloak, Azure and many others have a field in their dashboards about `Valid Redirect URLs` for auth code grant flow. This field is mandatory to not getting invalid redirect uri error. As far as I recall, you can not set this field with a port number which is a wild card(e.g. `127.0.0.1:*`) at least in Keycloak. Even if dashboard allowed it to set wild card as a port number, that would have been definitely insecure because Keycloak(or Azure) administrator opens to every port number in localhost for client. 

Therefore, this port number should be deterministic(as in all the other credentials exec plugins) and defaults to something(I set it to `8000` but open to suggestions).

Additionally, since port number is deterministic, Keycloak(and Azure) allows multiple valid redirect urls that can work in cases where if the first port number is not unused, falling back to the other one until it finds any unused port. That's the reason of why `CallbackPort` field is array not a single value.

Reply:
for more info: https://github.com/int128/kubelogin/blob/622dc5ba0bf31c6804289885500e9c723af012b9/pkg/cmd/authentication.go#L59

Reply:
That's a misunderstanding of the concept of native clients.

IdPs allow you to configure a redirect URL either http://127.0.0.1 or http://localhost - that is *without a port*. A random port is already assumed. Read more about it here - https://datatracker.ietf.org/doc/html/rfc8252

Please default to random ports.

Reply:
Updated to pass `127.0.0.1:0` as callback address if callback-port is not specified to pick a random unused port.